app/scripts/model/dataProxy.js

```diff
@@ -1279,6 +1280,94 @@ window.DataManagerForIviz = (function($, _) {
       getClinicalData: function(attribute_ids, isPatientAttributes) {
         return isPatientAttributes ? this.getPatientClinicalData(attribute_ids) :
           this.getSampleClinicalData(attribute_ids);
+      },
+      getAllGenePanelSampleIds: window.cbio.util.makeCachedPromiseFunction(
+        function (self, fetch_promise) {
+          var _map = {};
+          var asyncAjaxCalls = [];
+          var responses = [];
+          _.each(self.getCancerStudyIds(), function(_studyId) {
+            asyncAjaxCalls.push(
+              $.ajax({
+                url: window.cbioURL + 'api-legacy/genepanel/data',
+                contentType: "application/json",
+                data: ["profile_id=" + _studyId + "_mutations", "genes="].join("&"),
```

Comment:
I talked to Zack about it. It is better to do it this way instead of passing 600+ genes into it. And the API should be designed to return all gene panels under this study. I think a legit fix here is asking Zack (who wrote this API) to accept request with no "genes" parameter